src/scrollspy/scrollspy.ts

```diff
@@ -0,0 +1,44 @@
+import {Directive, ElementRef, Injectable, Input, OnDestroy, OnInit, Renderer} from 'angular2/core';
+
+@Directive({selector: '[ngbScrollSpy]', exportAs: 'ngbScrollSpy'})
+export class NgbScrollSpy implements OnDestroy, OnInit {
+  active: boolean;
+  @Input() ngbScrollSpy: string;
+
+  constructor(public elementRef: ElementRef, private scrollSpier: NgbScrollSpier) {}
+
+  ngOnDestroy(): void { this.scrollSpier.deregister(this); }
+
+  ngOnInit(): void { this.scrollSpier.register(this); }
+}
+
+@Directive({selector: '[ngbScrollTarget]', exportAs: 'ngbScrollTarget'})
+export class NgbScrollTarget implements OnInit {
+  constructor(private elementRef: ElementRef, private renderer: Renderer, private scrollSpier: NgbScrollSpier) {}
+
+  checkSpies(evt: Event) { this.scrollSpier.spies.forEach(spy => this.checkSpy(spy, evt)); }
+
+  checkSpy(spy: NgbScrollSpy, evt: Event) {
+    let element = this.elementRef.nativeElement;
+    let target = element.querySelector(spy.ngbScrollSpy);
+
+    if (!target) {
+      return;
+    }
+
+    this.renderer.setElementClass(
+        spy.elementRef.nativeElement, 'active',
+        target.offsetTop <= element.scrollTop && target.offsetTop + target.offsetHeight >= element.scrollTop);
+  }
+
+  ngOnInit(): void { this.renderer.listen(this.elementRef.nativeElement, 'scroll', (e) => this.checkSpies(e)); }
+}
+
+@Injectable()
+export class NgbScrollSpier {
+  spies = new Set<NgbScrollSpy>();
```

Comment:
It would be nice to use a WeakSet for this, but I assume we should stay away from that ES6 feature until we're supporting Edge and up.